devfiles/java-web-spring/devfile.yaml

```diff
@@ -0,0 +1,54 @@
+---
+apiVersion: 1.0.0
+metadata:
+  name: java-web-spring
+projects:
+  -
+    name: java-web-spring
+    source:
+      type: git
+      location: "https://github.com/spring-projects/spring-petclinic.git"
+components:
+  -
+    type: chePlugin
+    id: redhat/java/latest
+  -
+    type: dockerimage
+    alias: tools
+    image: registry.centos.org/che-stacks/centos-jdk8
+    command: ['sleep']
+    args: ['infinity']
+    env:
+      - name: JAVA_OPTS
+        value: "-XX:MaxRAMPercentage=50.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
+          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90
+          -Dsun.zip.disableMemoryMapping=true -Xms20m -Djava.security.egd=file:/dev/./urandom
+          -Duser.home=/home/user"
+      - name: MAVEN_OPTS
+        value: $(JAVA_OPTS)
+      - name: PS1
+        value: $(echo ${0})\\$
+    memoryLimit: 1024Mi
+    endpoints:
+      - name: '8080/tcp'
+        port: 8080
+    mountSources: true
+    volumes:
+      - name: m2
+        containerPath: /home/user/.m2
+commands:
```

Comment:
The same comments are actual as I mentioned above.
Either `clonePath` should be set or `workDir` modified.

Reply:
This looks like clone is broken. According to doc (https://redhat-developer.github.io/devfile/devfile#projects), when no `clonePath` defined, project name should be used as clone folder. However, here project is cloned to `spring-petclinic`, which is wrong. Looks like it's related to https://github.com/eclipse/che/issues/13796, see https://github.com/eclipse/che/issues/13796#issuecomment-509369498 comment. Some path is not set properly in Theia.

Reply:
I went through existed devfiles and realized that some uses `workdir` based on project name and some based on git location. We have to streamline everything.

Reply:
yes, we must retest all of them after https://github.com/eclipse/che-theia/pull/337 is merged

Reply:
fixed